pkgs/development/libraries/gtk+/3.0-immodules.cache.patch

```diff
@@ -0,0 +1,24 @@
+--- a/gtk/deprecated/gtkrc.c	2016-04-02 18:43:08.401663112 +0900
++++ b/gtk/deprecated/gtkrc.c	2016-04-02 18:29:19.927608592 +0900
+@@ -771,5 +771,20 @@
+   const gchar *var = g_getenv ("GTK_IM_MODULE_FILE");
+   gchar *result = NULL;
+ 
++  // check NIX_PROFILES paths.
++  const gchar *nixProfilesEnv = g_getenv ("NIX_PROFILES");
++  gchar *cachePath;
++  guint i;
++
++  if(nixProfilesEnv){
++    gchar **paths = g_strsplit(nixProfilesEnv, " ", -1);
++    for (i = 0; paths[i] != NULL; i++){
++      cachePath = g_build_filename(paths[i], "etc", "gtk-3.0", "immodules.cache", NULL);
++      if( g_file_test( cachePath, G_FILE_TEST_EXISTS) ){
++        result = g_strdup(cachePath);
++      }
```

Comment:
This leaks memory in a few places:
 * `paths`, just add a `g_strfreev(paths)` at the end.
 * `cachePath` needs to be freed as well.
 * What should happen when more than one profile contains the etc/gtk-3.0/immodules.cache` directory?

(When running programs in valgrind to check for memory leaks, it is nice if there are no false positives from system libraries.  Or more realistically, as few as possible.)